File: gcd and lcm.py
from itertools import count
import math
t=int(input())
def fun(n):
    count=0
    a=[]
    w=int(math.sqrt(n))
    for i in range(1,w+1):
        bs=n-i*i
        b=int(math.sqrt(bs))
        if(b*b==bs):
            a.append([i,b])
    c=[]
    b=[]
    for i in a:
        if i not in b:
            c.append(i)
            b.append([i[1],i[0]])
    for i in c:
        p=min(i[0], i[1])
        q=max(i[0],i[1])
        if(i[0]==0 or i[1]==0):
            count+=1
        else:
            for j in range(1, q):
                a=j
                b=a+p
                g=math.gcd(a,b)
                l=(a*b)//g
                if(g+l)==q:
                    count+=2
    return count                

for i in range(t):
    n=int(input())
    if n%2==0:
        print(fun(n))
    else:
        print(0)    

File: gcd and lcm.cpp
#include<iostream>
#include<bits/stdc++.h>
using namespace std;
bool check(long long j){
     if (ceil((double)sqrt(j)) == floor((double)sqrt(j))) return 1;
     return 0;
}

int main(){
int count;
cin>>count;

while(count-->0){

long long N;
cin>>N;

if(N%2!=0) {cout<<0<<endl; continue;}

long long sqrtn=sqrt(N);
vector<pair<long long, long long>> vp;

for (long long i = 0; i <= sqrtn+1/2; i++)
{ long long j=N- (i*i);
if(check(j)) {long long j1=sqrt(j); vp.push_back(make_pair(min(i,j1), max(i,j1)));}
}

int ans=0;

for(int i=0;i<vp.size()/2;i++){
long long a, b, g, l;
for(b=1;b<=vp[i].second;b++){
    // cout<<a;
    a=vp[i].first+b;
    g=__gcd(a,b);
    l=(a*b)/g;
    if(g+l==vp[i].second){
        if(a==b) ans+=1;
        else ans+=2;
    }
}

}
cout<<ans<<endl;
}
    return 0;
}